client/src/main/java/org/eclipse/hono/client/impl/CredentialsClientImpl.java

```diff
@@ -63,9 +59,16 @@ protected String createMessageId() {
     }
 
     @Override
-    protected CredentialsResult getResult(final int status, final JsonObject payload) {
-
-        return CredentialsResult.from(status, payload);
+    protected CredentialsResult<CredentialsObject> getResult(final int status, final String payload) {
+        try {
+            if (status == HTTP_OK) {
+                ObjectMapper om = new ObjectMapper();
```

Comment:
The Jackson docs recommend to re-use `ObjectMapper` instances. Maybe they are expensive to create?